wsr-tic-tac-toe/src/redux/reducers/index.ts

```diff
@@ -0,0 +1,137 @@
+import {
+    WinningScenarioActionTypes,
+    BoardState,
+    BoardStateActionTypes,
+    SquareState,
+    GAME_STATE,
+} from '../types';
+import {
+    CLEAR_WINNING_SCENARIOS,
+    CREATE_BOARD,
+    REMOVE_BOARD,
+    GENERATE_WINNING_SCENARIOS,
+    SELECT_SQUARE,
+    CLEAR_BOARD,
+    SET_GAME_STATE,
+} from '../actions';
+import { combineReducers } from 'redux';
+const initialWinningScenariosState: Array<Array<number>> = [];
+
+export function winningScenariosReducer(
+    state = initialWinningScenariosState,
+    action: WinningScenarioActionTypes
+) {
+    switch (action.type) {
+        case CLEAR_WINNING_SCENARIOS: {
+            return initialWinningScenariosState; // return initialWinningScenariosState
+        }
+        case GENERATE_WINNING_SCENARIOS: {
+            let boardSize = action.payload;
+            let WSh: Array<Set<number>> = []; // array of horizontal winning sets
```

Comment:
algorithm should be in a separate function.
Very good target for some nice clear unit tests too :)

Reply:
Should I create a new file "game-logic.ts" for these functions or write the function inside this reducer file ?